### ISSUE TITLE
Ability to categorise rules into fixable and unfixable

### ISSUE THREAD

**Issue by sowusu-ba** [sqlfluff/sqlfluff#5222]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

I believe it will be helpful to  be able to run 
```bash
sqlfluff fix --fixable
```
Which will apply this to all `sqlfluff fix` compatible rules.

### Use case

Currently, you'd have to list a lot of rules to get all sqlfluff fix compatible rules

### Dialect

Bigquery etc

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by fmms** [comment]
Awesome was thinking about that as well . Could be a huge performance improvement

**Comment by alanmcruickshank** [comment]
@sowusu-ba & @fmms - are you both thinking about this _purely_ from a performance perspective? i.e. we don't want the overhead of rules which can't generate fixes, if all we're interested in is fixing files.

In theory the result of `sqlfluff fix ...` and `sqlfluff fix --fixable ...` *should* be the same, because in both cases, only fixes would be generated by rules which are able to generate fixes.

If that _is_ what you're implying, I assume a better solution here would be to filter only to rules which support fixes *every* time `sqlfluff fix` is run anyway?

**Comment by fmms** [comment]
@alanmcruickshank yes my thinking was from a pure performance perspective. Running `sqlfluff fix` in our environment takes 45 minutes or so. In the output I have seen warnings by unfixable elements -- which someone might be interested in.
For our personal use case I'd be happy as well if fix only runs the fixable rules.

**Comment by alanmcruickshank** [comment]
I think it's a pretty reasonable default behaviour to only run fixable rules in you're running `fix`. It's worth pointing out that sometimes rules that are marked as being able to fix code, sometimes find unfixable scenarios (or at least one where the rule can't work out the "correct" scenario to fix to), so even with this change you might still see some unfixable warnings.

**Comment by sowusu-ba** [comment]
I just want to be able to group rules. There are so many rules and the team has done a great job of categorizing and documenting them on the website. Hwoever, if I want a list of all rules that are `sqlfluff fix` compatible, I'd have to list them one by one. Say I want to run `sqlfluff lint` but only want my linting to focus on those that can be fixed (all things being equal) by the linter. I don't want to `sqlfluff fix`, I just want to lint and see the results and get the output for further documentation. I currently have to list all the rules because there's no flag that separates fixable rules and non-fixable ones.

**Comment by alanmcruickshank** [comment]
@sowusu-ba - this is the part that I'd love to understand more. _All_ of the rules are resolvable by a person, so the thing I'm not clear on is why you'd only want to display the machine-fixable rules to a human user, unless it was part of applying the automatic fixes? What would you expect the user to do with the information which is different to the non-auto-fixable rules?

**Comment by sowusu-ba** [comment]
@alanmcruickshank - for the machine-fixable, we have a github action that suggests fixes as comments in our PR. However, for the ones that need to be resolved by a person, we run another step that shows the errors to the user. We want an easy way to separate the two rules. For instance, in the second step that shows errors to the user, we just want to have something like `sqlfluff lint --exclude --machine-fixable-rules`. We are currently listing these rules one by one for each step. I hope that makes sense.

**Comment by alanmcruickshank** [comment]
@sowusu-ba - I think I understand.

One edge case I'd love to get clear on. There are some rules which are sometimes fixable but sometimes not. For example `LT05` (long lines), will suggest a fix if it can work one out, but in cases where the line is too long and it can't find an appropriate fix will still return a linting issue, just not a "fixable" one.

From your example use case I assume you would want `LT05` to run in both cases. You'd want it to run both in the `fixable` case (so the automatic fixes are generated), but _also_ in the `unfixable` case so that the non-fixable situations are still pointed out to the user?

On alternative solution here, would be to generate both sets of fixes in the same pass, but provide an alternative way of saving them separately somehow. I'm not sure how to do that in a way that best fits with your situation but it might be cleaner.

For the use case described by @fmms - I think it's clearer that we'd filter to any rule which _might_ be able to fix a situation (which is already on rules as their `is_fix_compatible` attribute), even if occasionally they might fail to generate a fix in some situations. It is worth pointing out however that if this is for performance reasons, the slowest rules are `LT01`, `LT02`, `LT04`, `LT05` and `ST05` which are all `fixable` rules, so eliminating the non-fixable ones won't save that much time on a long run.

**Comment by peterbud** [comment]
Should we keep this issue still open?
